app/src/main/java/org/sil/gatherwords/EntryActivity.java

```diff
@@ -297,4 +315,85 @@ public PlayButton(Context ctx) {
         }
     }
 
+    private static class SaveRecordingTask extends AsyncTask<String, Void, Void>{
+        private AppDatabase db;
+        private WordDao wd;
+        private WeakReference<EntryActivity> entryActivityRef;
+        private long wordId;
+
+        SaveRecordingTask(EntryActivity activity) {
+            entryActivityRef = new WeakReference<>(activity);
+            db = AppDatabase.get(activity.getApplicationContext());
+            wd = db.wordDao();
+            EntryFragment fragment = (EntryFragment) activity.getCurrentFragment();
+            wordId = fragment.getWordId();
+        }
+
+        @Override
+        protected Void doInBackground(String... strings) {
+            EntryActivity activity = entryActivityRef.get();
+            String filename = activity.getFileName();
+            if (activity == null || filename == null) {
+                return null;
+            }
+            byte[] audio = null;
+            try {
+                InputStream is = new FileInputStream(filename);
+                audio = new byte[is.available()];
```

Comment:
Try `ByteArrayOutputStream` instead so we do not need line 344.

Reply:
I'm seeing ways to read a file with `InputStream` and `BufferedInputStream` but not away with `ByteArrayOutputStream` or `ByteArrayInputStream`

Can you point me to a source?